### PR TITLE
Proto update and Unknown6 Fix

### PR DESCRIPTION
**Pull request by dmadisetti**
OK.

So this worked, and then it didn't. Then it worked again. Something is screwy- but not fully sure what. 

For usage, requires the shared library (encrypt.dll or libencrypt.so). For copyright reasons, not provided. But with some searching you might be able to find it. 

I'm going to bed, but if someone wants to try and debug this- feel free.

### CODE REVIEW

File: pogo/util.py [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+OR OTHER DEALINGS IN THE SOFTWARE.
+
+Author: tjado <https://github.com/tejado>
```
Comment (rubenvereecken) [inline pogo/util.py]: I'm pretty sure this is not tejado's code, I followed the discussion for a bit.

File: pogo/util.py [hunk 2]
```diff
@@ -13,10 +40,58 @@ def f2h(float):
...
+    baseHash = xxhash.xxh64(authTicket.SerializeToString(), seed=0x1B845238).intdigest()
+    return xxhash.xxh64(payload.SerializeToString(), seed=baseHash).intdigest() # for request in payload]
+
+
+# Assuming the encrypt.dll file floating around out there
+def hashSignature(signature, libraryPath):
+    serialized = signature.SerializeToString()
+    size = len(serialized)
+
+    library = ctypes.cdll.LoadLibrary(libraryPath)
```
Comment (rubenvereecken) [inline pogo/util.py]: this is good for now but I guess we should look into loading from `LD_LIBRARY_PATH`

**Comment by rubenvereecken** [comment]
Thank you so much for merging in the newest changes. I'll have a decent look at `libencrypt.so` once I get the time.

**Comment by rubenvereecken** [comment]
Eh could you maybe leave a tiny note in the Readme on where to find the required library?

**Comment by progmatik** [comment]
I get it work but need to change location every time it asks to find forts. So I wrote a script based on demo.py every time I walk to a fort I get his location and pass it to arg

**Pull request merged**